obspy/signal/polarization.py

```diff
@@ -86,48 +87,482 @@ def eigval(datax, datay, dataz, fk, normf=1):
-    # dleigenv1 = dleigenv1[np.size(fk) // 2:(np.size(dleigenv1) -
-    #        np.size(fk) / 2)]
 
-    leigenv2_add = np.append(
-        np.append(
-            [leigenv2[0]] * (np.size(fk) // 2),
-            leigenv2), [leigenv2[np.size(leigenv2) - 1]] * (np.size(fk) // 2))
+    leigenv2_add = np.append(np.append([leigenv2[0]] * (np.size(fk) // 2),
+                                       leigenv2),
+                             [leigenv2[np.size(leigenv2) - 1]] *
+                             (np.size(fk) // 2))
     dleigenv2 = signal.lfilter(fk, 1, leigenv2_add)
     dleigenv[:, 1] = dleigenv2[len(fk) - 1:]
-    # dleigenv2 = dleigenv2[np.size(fk) // 2:(np.size(dleigenv2) -
-    #        np.size(fk) / 2)]
 
-    leigenv3_add = np.append(
-        np.append(
-            [leigenv3[0]] * (np.size(fk) // 2), leigenv3),
-        [leigenv3[np.size(leigenv3) - 1]] * (np.size(fk) // 2))
+    leigenv3_add = np.append(np.append([leigenv3[0]] * (np.size(fk) // 2),
+                                       leigenv3),
+                             [leigenv3[np.size(leigenv3) - 1]] *
+                             (np.size(fk) // 2))
     dleigenv3 = signal.lfilter(fk, 1, leigenv3_add)
     dleigenv[:, 2] = dleigenv3[len(fk) - 1:]
-    # dleigenv3 = dleigenv3[np.size(fk) // 2:(np.size(dleigenv3) -
-    #        np.size(fk) / 2)]
 
-    rect_add = np.append(
-        np.append([rect[0]] * (np.size(fk) // 2), rect),
-        [rect[np.size(rect) - 1]] * (np.size(fk) // 2))
+    rect_add = np.append(np.append([rect[0]] * (np.size(fk) // 2), rect),
+                         [rect[np.size(rect) - 1]] * (np.size(fk) // 2))
     drect = signal.lfilter(fk, 1, rect_add)
     drect = drect[len(fk) - 1:]
-    # drect = drect[np.size(fk) // 2:(np.size(drect3) - np.size(fk) // 2)]
 
-    plan_add = np.append(
-        np.append([plan[0]] * (np.size(fk) // 2), plan),
-        [plan[np.size(plan) - 1]] * (np.size(fk) // 2))
+    plan_add = np.append(np.append([plan[0]] * (np.size(fk) // 2), plan),
+                         [plan[np.size(plan) - 1]] * (np.size(fk) // 2))
     dplan = signal.lfilter(fk, 1, plan_add)
     dplan = dplan[len(fk) - 1:]
-    # dplan = dplan[np.size(fk) // 2:(np.size(dplan) - np.size(fk) // 2)]
 
     return leigenv1, leigenv2, leigenv3, rect, plan, dleigenv, drect, dplan
+
+
+def flinn(stream, noise_thres=0):
+    """
+    Computes the azimuth, incidence, rectilinearity and planarity after the
+    eigenstructure decomposition method of [Flinn1965b]_.
+
+    :param stream: ZNE sorted trace data
+    :type stream: :class:`~obspy.core.stream.Stream`
+    :param noise_tresh: Variance of the noise sphere; data points are excluded
+        when falling within the sphere with radius sqrt(noise_thres),
+        default is set to 0.
+    :type noise_thres: float
+    :returns:  azimuth, incidence, rectilinearity, and planarity
+    """
+    Z = []
+    N = []
+    E = []
+    comp, npts = np.shape(stream)
+
+    for i in range(0, npts):
```

Comment:
Not easily, length is flexible. This means we would need two passes of the for loop